schema.go

```diff
@@ -6,6 +6,18 @@ import (
 	"github.com/vektah/gqlparser/v2/ast"
 )
 
+// Operation is a GraphQL operation type.
+type Operation string
```

Comment:
I don't see this type offering much benefit, as it is always cast to a string before use and should always be coming from a valid graphql query. The swithc in `WithOperationType` could handle that case without the need for the type wrapper

Reply:
I think this type could be useful for replacing `ParentType` in the struct below. But that requires a bigger change, that I think would be nice to have in another PR. What do you think about this? 😄

```go
// plan.go

// QueryPlanStep is a single execution step
type QueryPlanStep struct {
	ServiceURL     string
	ServiceName    string
	ParentType     string --> Operation
	SelectionSet   ast.SelectionSet
	InsertionPoint []string
	Then           []*QueryPlanStep
}
```

Reply:
If we're going to make that change I'd suggest making it all in it's own PR instead of a partial change in this one.

Reply:
I removed it for now 😄